scaaml/capture/input_generators/attack_point_iterator.py

```diff
@@ -152,3 +153,30 @@ def __iter__(self):
 
     def get_generated_keys(self) -> List[str]:
         return [self._name]
+
+
+class AttackPointIteratorRepeat(AttackPointIterator):
+    """
+    Attack point iterator class that iterates 
+    over a configuration a repeated amount of times.
+    """
+
+    def __init__(self, operation: str, repetitions: int,
+                 configuration: Dict) -> None:
+        """Initialize the repeated iterate."""
+        assert "repeat" == operation
+        self._repetitions = repetitions
+        self._configuration_iterator = build_attack_points_iterator(
+            configuration)
+        self._len = repetitions * len(self._configuration_iterator)
+
+    def __len__(self) -> int:
+        return self._len
+
+    def __iter__(self):
+        return iter(
+            list(iter(self._configuration_iterator))
```

Comment:
Be careful, calling `list()` here will trigger pre-computing all the iterator values in-memory. Depending on the configuration passed as a parameter, that could crash

Reply:
```
def __iter__(self):
  return self

def __next__(self):
  for _ in range(self._repetitions):
    for attack_points in self._configuration_iterator:
      yeild attack_points
```

Reply:
Please add the following test to test that the memory blow-up mentioned by JM does not happen:

```
def test_repeat_memory():
    config = {
        "operation": "repeat",
        "repetitions": 1_000_000_000_000,
        "configuration": {
            "operation": "constants",
            "name": "key",
            "values": [1, 2, 3]
        }
    }

    long_iterator = iter(build_attack_points_iterator(config))
    assert next(long_iterator) == { "key": 1 }
    assert next(long_iterator) == { "key": 2 }
    assert next(long_iterator) == { "key": 3 }
    assert next(long_iterator) == { "key": 1 }
    assert next(long_iterator) == { "key": 2 }
    assert next(long_iterator) == { "key": 3 }
    assert next(long_iterator) == { "key": 1 }
```